krummstab/__init__.py

```diff
@@ -365,6 +367,197 @@ def unsupported_marking_mode_error(_the_config: config.Config) -> None:
+                if sheet_name not in students_marks[email]:
+                    students_marks[email][sheet_name] = {}
+                students_marks[email][sheet_name] = exercises
+                for exercise in exercises:
+                    if exercise not in graded_sheet_names[sheet_name]:
+                        graded_sheet_names[sheet_name].append(exercise)
+
+    bold_format = workbook.add_format({'bold': True})
+    worksheet.write(3, 0, "First Name", bold_format)
+    worksheet.write(3, 1, "Last Name", bold_format)
+    worksheet.write(3, 2, "Total Points", bold_format)
+    worksheet.write(1, 0, "Max Points", bold_format)
+    worksheet.write(2, 0, "Average", bold_format)
+
+    all_sheet_names = _the_config.max_points_per_sheet.keys()
+    student_start_row = 4
+    student_end_row = student_start_row + len(students_marks) - 1
+    col = 3
+    for sheet_name in all_sheet_names:
+        worksheet.write(0, col, sheet_name, workbook.add_format({'bold': True, "left": 1}))
+        max_points_value = _the_config.max_points_per_sheet.get(sheet_name)
+        worksheet.write(1, col, max_points_value, workbook.add_format({"left": 1}))
+        if sheet_name in graded_sheet_names:
+            avg_range = xl_range(student_start_row, col, student_end_row, col)
+            worksheet.write_formula(2, col, f"=AVERAGE({avg_range})", workbook.add_format({"left": 1}))
+            col += 1
+            for exercise in sorted(graded_sheet_names[sheet_name]):
+                worksheet.write(0, col, exercise.replace("exercise", "task"), bold_format)
+                col += 1
+        else:
+            col += 1
+
+    workbook.close()
+
+
+def summarize_points_per_sheet(_the_config: config.Config) -> None:
+    email_to_name = {}
+    for team in _the_config.teams:
+        for first_name, last_name, email in team:
+            email_to_name[email] = (first_name, last_name)
+
+    workbook = xlsxwriter.Workbook("Points_Summary_Report.xlsx")
+    worksheet = workbook.add_worksheet("Points Summary")
+
+    marks_files = args.marks_dir.glob("points_*_individual.json")
+    students_marks = {}
+    graded_sheet_names = set()
+
+    for file in marks_files:
+        with open(file, 'r') as f:
+            data = json.load(f)
+            sheet_name = data["adam_sheet_name"]
+            marks = data["marks"]
+            graded_sheet_names.add(sheet_name)
+
+            for email, mark in marks.items():
+                if email not in students_marks:
+                    students_marks[email] = {}
+                students_marks[email][sheet_name] = mark
+
+    bold_format = workbook.add_format({'bold': True})
+    worksheet.write(3, 0, "First Name", bold_format)
+    worksheet.write(3, 1, "Last Name", bold_format)
+    worksheet.write(3, 2, "Total Points", bold_format)
+    worksheet.write(1, 0, "Max Points", bold_format)
+    worksheet.write(2, 0, "Average", bold_format)
+
+    all_sheet_names = _the_config.max_points_per_sheet.keys()
```

Comment:
I ran out of time before the meeting and stopped reviewing here. If you can split the function into smaller pieces, it would also help with reviewing. For example, you could add one function for each "feature" added to the work sheet, like `add_student_marks`, `add_average_column`, `add_plagiarism_conditional_formatting`, etc.